src/collaboration/pull-protocol.js

```diff
@@ -180,15 +172,120 @@ module.exports = class PullProtocol {
     return this._cachedPeerId
   }
 
+  _remote () {
+    const output = pushable()
+    let eager = true
+
+    return {
+      output,
+      send (msg) {
+        // [clock, startLazy, startEager, isLocalPinner]
+        output.push(encode(msg))
+      },
+      sendClock (clock) {
+        this.send([clock])
+      },
+      setLazyMode () {
+        if (eager) {
+          eager = false
+          this.send([null, true])
+        }
+      },
+      setEagerMode () {
+        if (!eager) {
+          eager = true
+          this.send([null, false, true])
+        }
+      },
+      init (clock, isLocalPinner) {
+        this.send([clock, null, null, isLocalPinner])
+      },
+      end (err) {
+        output.end(err)
+      }
+    }
+  }
+
+  _waitTimers (dbg) {
+    const receiveTimeoutMS = this._options.receiveTimeoutMS
+    const queue = new Queue({ concurrency: 1 })
+    const timers = new Map()
+    let running = true
+
+    const localHasClock = (clock) => {
+      const localClock = this._clocks.getFor(this._peerId())
+      return vectorclock.doesSecondHaveFirst(clock, localClock)
+    }
+
+    return {
+      onClock (clock, onTimeout) {
+        queue.add(() => {
```

Comment:
Is a promise queue necessary here? This function is synchronous, so it will execute to completion. Or do you need to synchronise with other asynchronous functions also executing in this queue?

Reply:
Good point, much simpler without the promise queue